go/config_store.go

```diff
@@ -0,0 +1,7 @@
+package configu
+
+type IConfigStore interface {
+	Get(queries []ConfigStoreQuery) ([]Config, error)
+	Set(configs []Config) error
+	GetType() string
```

Comment:
@jamesL92 Should we consider adding this method across all SDKs? Or os it essentially serves the same purpose as a public property in Go?

Reply:
Essentially the equivalent of a public property in Go. There's also a public `Type` variable in the generated types [here](https://github.com/jamesL92/configu/blob/feat/go-implementation/go/generated.go#L214), but couldn't find a "clean" way of making use of it for concrete config stores (since the concept of subtyping doesn't translate well to Go). Interfaces are typically the idiomatic way to do this sort of thing in Golang.

For most OO languages (python/TS currently, but in future things like Java/C#) there's not really any benefit in adding this, although I think Rust will need a similar thing when that SDK is created/implemented.